src/main/java/mate/academy/Main.java

```diff
@@ -1,16 +1,99 @@
+    private static final int CINEMA_HALL_NUMBER_TWO = 1;
+    private static final int CINEMA_HALL_NUMBER_THREE = 2;
+    private static final LocalDate DATE_24_12_25 = LocalDate.of(2024, 12, 25);
+    private static final LocalDate DATE_24_12_27 = LocalDate.of(2024, 12, 27);
+    private static final int MOVIE_SESSION_ONE = 0;
+
     public static void main(String[] args) {
-        MovieService movieService = null;
 
-        Movie fastAndFurious = new Movie("Fast and Furious");
-        fastAndFurious.setDescription("An action film about street racing, heists, and spies.");
-        movieService.add(fastAndFurious);
-        System.out.println(movieService.get(fastAndFurious.getId()));
+        // 1. MovieService realisations (add, get, getAll).
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
+        List<Movie> movies = createSampleMovies();
+
+        for (Movie movie : movies) {
+            movieService.add(movie);
+        }
+
+        System.out.println(movieService.get(movies.get(FAST_AND_FURIOUS_MOVIE).getId()));
         movieService.getAll().forEach(System.out::println);
+
+        //2. CinemaHallService realisations (add, get, getAll).
+        CinemaHallService cinemaHallService = (CinemaHallService) injector.getInstance(
+                CinemaHallService.class);
+        List<CinemaHall> cinemaHalls = createSampleCinemaHalls();
+
+        for (CinemaHall cinemaHall : cinemaHalls) {
+            cinemaHallService.add(cinemaHall);
+        }
+
+        System.out.println(cinemaHallService.get(cinemaHalls.get(CINEMA_HALL_NUMBER_ONE).getId()));
+        cinemaHallService.getAll().forEach(System.out::println);
+
+        //3. MovieSessionService realisations (add, get, findAvailableSessions).
+        MovieSessionService movieSessionService = (MovieSessionService) injector.getInstance(
+                MovieSessionService.class);
+
+        List<MovieSession> movieSessions = List.of(
+                new MovieSession(movies.get(FAST_AND_FURIOUS_MOVIE),
+                        cinemaHalls.get(CINEMA_HALL_NUMBER_ONE), DATE_24_12_25),
+                new MovieSession(movies.get(THE_GODFATHER_MOVIE),
+                        cinemaHalls.get(CINEMA_HALL_NUMBER_ONE), DATE_24_12_25),
+                new MovieSession(movies.get(FAST_AND_FURIOUS_MOVIE),
+                        cinemaHalls.get(CINEMA_HALL_NUMBER_THREE), DATE_24_12_25),
+                new MovieSession(movies.get(THE_DARK_KNIGHT_MOVIE),
+                        cinemaHalls.get(CINEMA_HALL_NUMBER_TWO), DATE_24_12_27),
+                new MovieSession(movies.get(THE_GODFATHER_MOVIE),
+                        cinemaHalls.get(CINEMA_HALL_NUMBER_THREE), DATE_24_12_27)
+
+        );
+
+        for (MovieSession movieSession : movieSessions) {
+            movieSessionService.add(movieSession);
+        }
+
+        System.out.println(movieSessionService.get(movieSessions.get(MOVIE_SESSION_ONE).getId()));
```

Comment:
Directly calling `.get()` on an `Optional` without checking if a value is present can lead to `NoSuchElementException`. Consider using `ifPresent` or `orElseThrow` to handle absent values more gracefully.